### ISSUE TITLE
[Proposal] TouchEffect

### ISSUE THREAD

**Issue by brminnick** [CommunityToolkit/Maui#86]
# TouchEffect

* [x] Proposed
* [x] Prototype
* [x] Implementation
  * [ ] iOS Support
  * [x] Android Support
  * [ ] macOS Support
  * [ ] Windows Support
* [ ] Unit Tests: Not Started
* [ ] Sample: Not Started
* [ ] Documentation: Not Started

## Summary
[summary]: #summary

Adds additional touch events to any VisualElement

## Detailed Design
[design]: #detailed-design

### TouchEffect.shared.cs

```cs
public class TouchEffect : RoutingEffect
{
  public event EventHandler<TouchStatusChangedEventArgs> StatusChanged;
  public event EventHandler<TouchStateChangedEventArgs> StateChanged;
  public event EventHandler<TouchInteractionStatusChangedEventArgs> InteractionStatusChanged;
  public event EventHandler<HoverStatusChangedEventArgs> HoverStatusChanged;
  public event EventHandler<HoverStateChangedEventArgs> HoverStateChanged;
  public event EventHandler<TouchCompletedEventArgs> Completed;
  public event EventHandler<LongPressCompletedEventArgs> LongPressCompleted;
  
  public static readonly BindableProperty IsAvailableProperty;
  public static readonly BindableProperty ShouldMakeChildrenInputTransparentProperty;
  public static readonly BindableProperty CommandProperty;
  public static readonly BindableProperty LongPressCommandProperty;
  public static readonly BindableProperty CommandParameterProperty;
  public static readonly BindableProperty LongPressCommandParameterProperty;
  public static readonly BindableProperty LongPressDurationProperty;
  public static readonly BindableProperty StatusProperty;
  public static readonly BindableProperty StateProperty;
  public static readonly BindableProperty InteractionStatusProperty;
  public static readonly BindableProperty HoverStatusProperty;
  public static readonly BindableProperty HoverStateProperty;
  public static readonly BindableProperty NormalBackgroundColorProperty;
  public static readonly BindableProperty HoveredBackgroundColorProperty;
  public static readonly BindableProperty PressedBackgroundColorProperty;
  public static readonly BindableProperty NormalOpacityProperty;
  public static readonly BindableProperty HoveredOpacityProperty;
  public static readonly BindableProperty PressedOpacityProperty;
  public static readonly BindableProperty NormalScaleProperty;
  public static readonly BindableProperty HoveredScaleProperty;
  public static readonly BindableProperty PressedScaleProperty;
  public static readonly BindableProperty NormalTranslationXProperty;
  public static readonly BindableProperty HoveredTranslationXProperty;
  public static readonly BindableProperty PressedTranslationXProperty;
  public static readonly BindableProperty NormalTranslationYProperty;
  public static readonly BindableProperty HoveredTranslationYProperty;
  public static readonly BindableProperty PressedTranslationYProperty;
  public static readonly BindableProperty NormalRotationProperty;
  public static readonly BindableProperty HoveredRotationProperty;
  public static readonly BindableProperty PressedRotationProperty;
  public static readonly BindableProperty NormalRotationXProperty;
  public static readonly BindableProperty HoveredRotationXProperty;
  public static readonly BindableProperty PressedRotationXProperty;
  public static readonly BindableProperty NormalRotationYProperty;
  public static readonly BindableProperty HoveredRotationYProperty;
  public static readonly BindableProperty PressedRotationYProperty;
  public static readonly BindableProperty AnimationDurationProperty;
  public static readonly BindableProperty AnimationEasingProperty;
  public static readonly BindableProperty PressedAnimationDurationProperty;
  public static readonly BindableProperty PressedAnimationEasingProperty;
  public static readonly BindableProperty NormalAnimationDurationProperty;
  public static readonly BindableProperty NormalAnimationEasingProperty;
  public static readonly BindableProperty HoveredAnimationEasingProperty;
  public static readonly BindableProperty PulseCountProperty;
  public static readonly BindableProperty IsToggledProperty;
  public static readonly BindableProperty DisallowTouchThresholdProperty;
  public static readonly BindableProperty NativeAnimationProperty;
  public static readonly BindableProperty NativeAnimationColorProperty;
  public static readonly BindableProperty NativeAnimationRadiusProperty;
  public static readonly BindableProperty NativeAnimationShadowRadiusProperty;
  public static readonly BindableProperty NormalBackgroundImageSourceProperty;
  public static readonly BindableProperty HoveredBackgroundImageSourceProperty;
  public static readonly BindableProperty PressedBackgroundImageSourceProperty;
  public static readonly BindableProperty BackgroundImageAspectProperty;
  public static readonly BindableProperty NormalBackgroundImageAspectProperty;
  public static readonly BindableProperty HoveredBackgroundImageAspectProperty;
  public static readonly BindableProperty PressedBackgroundImageAspectProperty;
  public static readonly BindableProperty ShouldSetImageOnAnimationEndProperty;
}
```

## Usage Syntax
[usage]: #usage-syntax

```xml
<Button TouchEffect.PressedScale="3">
</Button>
```

### C# Usage

```cs
var button = new Button();
TouchEffect.SetPressedScale(button, 3);
```

**Comment by mmendoza-apa** [comment]
Any updates on this?

Really looking forward to use the LongPress behavior!

**Comment by pictos** [comment]
Hey, @mmendoza-apa this is planned to land on v1.1.0

**Comment by naweed** [comment]
Hi @pictos , when is 1.1.0 targetted for release?

**Comment by pictos** [comment]
@naweed we don't have a date to share

**Comment by JohnHDev** [comment]
@pictos can I confirm if/when this is done it will be a Behaviour, not an effect, right?
We have started investigating what is needed to migrate to .NET Maui and we have dependencies that we either want to remove or migrate as well. We have a long press effect we don't want to migrate (unless we have to), we would rather go down the supported route with the replacement for TouchEffect.

**Comment by bijington** [comment]
@JohnHDev yes that is correct. Much like the `IconTintColor` and `SelectAllTextBehavior` implementations we will be moving from the old Effects API over to using Behaviors.

I don't know if @pictos but if not would you fancy the challenge of adding this in?

**Comment by beeradmoore** [comment]
Was just looking at planning ahead for my XF -> MAUI app migration which is planned for much further in the future. I use `NativeAnimation` a lot, I find it gives some nice polish. I don't think I can help contribute but I like seeing that the proposal is up at least.

**Comment by smardine** [comment]
Hello, is there any news about this?
I can't find it in current version 1.2
Hope it will be implement soon 🤞

**Comment by mantasdaskev** [comment]
> Hello, is there any news about this? I can't find it in current version 1.2 Hope it will be implement soon 🤞

Hello, I am wondering about that too. I have  few controls that depends on it and after migrating to MAUI I had to comment it out and release is now suspended.

**Comment by pictos** [comment]
Hey people, don't have a Date for you right now, but I'll jump on this one right after I finish the status bar implementation (should have a PR during this week). If the port goes well we can have this soonsh

**Comment by smardine** [comment]
> Hey people, don't have a Date for you right now, but I'll jump on this one right after I finish the status bar implementation (should have a PR during this week). If the port goes well we can have this soonsh

🤞😇

**Comment by KSemenenko** [comment]
Here is the most important point, in the experience of XF, that the ListView and CollectionView gestures did not allow to process all the other gestures.

And of course multiple gesture handlers are also needed

**Comment by LennoxP90** [comment]
looking forward to not having to continue using XCT MAUI Compat

**Comment by jonakirke94** [comment]
Looking foward to this as well!

In the meanwhile I'm wondering if anyone has gotten to Maui Compat version to work with TouchEffect?

I've opened an issue in the XamarinCommunityToolkit repo [here](https://github.com/xamarin/XamarinCommunityToolkit/issues/1905) but haven't found a solution yet :)

**Comment by LennoxP90** [comment]
Is there an ETA on this?

**Comment by KieranMaclagan** [comment]
Really looking forward to this, especially for the long press.

**Comment by KSemenenko** [comment]
@pictos @brminnick is there any design decision about using behavior or effect? 
I found old pr https://github.com/xamarin/Xamarin.Forms/pull/9740/ for it.

Maybe we can make decision about it and I can start implementation?

**Comment by pictos** [comment]
@KSemenenko this is in WIP mode, I'm working this. Thanks

**Comment by tranb3r** [comment]
Is there any progress for the long press support?

**Comment by fredericoregateiro** [comment]
Looks like long press take long time to implement 😄

**Comment by sottnik** [comment]
I don't understand. Is it a technical challenge or simply not interesting enough to be implemented?

**Comment by bijington** [comment]
> I don't understand. Is it a technical challenge or simply not interesting enough to be implemented?

We all dedicate our own free time to do this stuff so most likely nobody has found enough of that time to dedicate to this. If it is important to the community and your projects then please do consider contributing the implementation

**Comment by Hooterr** [comment]
I've got to say, I don't understand this approach at all. Something as crucial as TouchEffect and all other essential bits and pieces are not a part of MAUI itself but rather are "outsourced" to this community toolkit where they lay almost untouched for, in this case 1,5 yrs. At the same time, MAUI repo is swarmed with bugs that also lay untouched for many months or years. What is actually going on here? Is the MAUI team purposely understaffed? Does Microsoft care at all about this project? They sure seem to want to take credit for it, just like they did take credit for MediaElement on Twitter and had to delete the original tweet. 

To be honest, I don't understand the move to MAUI either. We had so much stuff implemented in Xamarin.Forms. Most of the bugs were either found a workaround for or fixed. But this... is just a total step or maybe even two steps backwards... 

As you can imagine we don't use MAUI for fun but almost exclusively for work. And when we have to spend time fixing bugs in the framework or implementing missing features it's harder and harder to convince the management to stick to this framework. All the advantages of .NET and MAUI are canceled out by its poor quality and missing features. If the plan is for MAUI to _eventually_ get good then by that time there will be hardly any people left that use it which would be a great shame.

**Comment by naweed** [comment]
That's exactly the reason why I moved away from Maui.

**Comment by Kalyxt** [comment]
If you are building bigger app not just few screen fun homemade product, you have to have controls and features from 3rd party suppliers like syncfusion, grial etc. It was like that in Xamarin forms times and its like that now too. Although you keep away from them as much as possible and use them only in must situations and if there appears option to replace it by open source alternative from community toolkit or other contributors, you try to switch it. Eventually you will be not dependant on paid 3rd party, but as long as you choose XF or MAUI you have to count with this.

**Comment by VladislavAntonyuk** [comment]
@Hooterr instead of writing such long comments you could already implement TouchEffect for at least one platform. As @bijington mentioned, we spend our free time delivering new features and bugfixes for this library. The main word in the name of this library is Community.

**Comment by Hooterr** [comment]
@VladislavAntonyuk that's right. So how can we expect our employers to allow us to spend company time paid time to fix things in a _Community_ project? Then the obvious answer is for us to use something else that doesn't need "fixing" from the start. Last time I checked MAUI is not a community project but a Microsoft product. The decision to move essential features to a _Community_ project is questionable, to say the least.

Honestly, in what universe is MAUI a competitive offer if the first thing you need to do with it is to fix numerous bugs and implement pretty basic features? Time spent doing that easily cancels out any benefits of using it and the frustration is just not worth it anymore.

I mean no disrespect to people for whom this is a passion project, but for the majority of us it's just a job. 

Gonna agree with @naweed here. It's probably time to move on.

**Comment by bijington** [comment]
Your company has likely made money from using this free toolkit so justifying providing some effort to help fix things really shouldn't be a debate.

**Comment by fredericoregateiro** [comment]
For me, I don't contribute more because I lack knowledge on the different platforms.

The work done in this project is spectacular, but sometimes it seems that the MAUI team doesn't develop some features waiting for the community to do it dotnet/maui#8675 

In my opinion this would be a feature that should be on the MAUI platform, since day 1, and should not be in an external community project.

**Comment by bijington** [comment]
> For me, I don't contribute more because I lack knowledge on the different platforms.
> 
We are always happy to assist wherever we can in learning about the other platforms. In truth we really appreciate any contributions even if they are only partial.
> 
> The work done in this project is spectacular, but sometimes it seems that the MAUI team doesn't develop some features waiting for the community to do it dotnet/maui#8675 
> 
Thank you. In truth with the MAUI team, they are massively understaffed which is sadly their burden but not their fault. This toolkit is aimed at both lifting some of that burden and also in the hope that features can be shipped quicker. Being part of .NET now means stricter rules around adding new features to the framework. 
> 
> In my opinion this would be a feature that should be on the MAUI platform, since day 1, and should not be in an external community project.

I am sure we could all agree more could be in the framework but given they are limited in numbers I think the team do a fantastic job with what they have.

**Comment by jfversluis** [comment]
As much as I appreciate a good discussion, doing that in here isn't really helping keeping this proposal on track. Let's please get back to discussing the actual feature here. Thank you!

**Comment by mirceamuresanse** [comment]
> @pictos can I confirm if/when this is done it will be a Behaviour, not an effect, right? We have started investigating what is needed to migrate to .NET Maui and we have dependencies that we either want to remove or migrate as well. We have a long press effect we don't want to migrate (unless we have to), we would rather go down the supported route with the replacement for TouchEffect.

May I ask what do you mean by "we would rather go down the supported route with the replacement for TouchEffect"? Is there an alternative for the TouchEffect in Maui?

**Comment by VladislavAntonyuk** [comment]
We are trying to replace effects with behaviors. effects are like renderers in terms of performance, so if possible you shouldn't use them.

**Comment by pictos** [comment]
@mirceamuresanse, in short terms, we're changing effects to be behaviors, since there will not be further improvements in that area in .net maui. AFAIK, there's no solution for that on the main framework yet... This is WIP mode here

**Comment by KSemenenko** [comment]
Maybe we can do just gesture recognized? I did it for xamarin some time ago, and do we we relay need that effect?
I totally get all concerns about this functionality, but probably all of us must wait for some architectural solution.

**Comment by BernhardPollerspoeck** [comment]
about 1.5 yeasr later this pretty important feature is still open...   
i found a post here from Wenyang. with a partial solution.. [https://learn.microsoft.com/en-us/answers/questions/900859/long-press-in-net-maui](https://learn.microsoft.com/en-us/answers/questions/900859/long-press-in-net-maui)

issnt this at least the start of what is required? if so, where is the issue its not in yet? is there any way to assist here? please post clear instructions what you require from the 'community'.

**Comment by bijington** [comment]
> about 1.5 yeasr later this pretty important feature is still open...   
> 
> i found a post here from Wenyang. with a partial solution.. [https://learn.microsoft.com/en-us/answers/questions/900859/long-press-in-net-maui](https://learn.microsoft.com/en-us/answers/questions/900859/long-press-in-net-maui)
> 
> 
> 
> issnt this at least the start of what is required?

I can't comment with too much knowledge but it certainly looks like a good starting point

> if so, where is the issue its not in yet? 

I suspect this was initially held up by the fact that effects were announced to become obsolete so there were learnings to be done on how to overcome that. Then as with everything else time is usually the issue. If you are willing to take this on then that would be great!

> is there any way to assist here?

We would welcome the implementation if you are willing 

> please post clear instructions what you require from the 'community'.

We have a contributing guide here which covers what is required https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md and for the general implementation this issue covers the API surface that we were aiming to bring from the Xamarin Community Toolkit, as you can see it is pretty big.

**Comment by JkPersson** [comment]
Any news on this one? When can we expect it?

**Comment by kphillpotts** [comment]
I am doing some work on this today - Expect to have a decent test branch next week if you are keen to test

**Comment by JkPersson** [comment]
Thanks - looking forward to seeing it.

**Comment by albilaga** [comment]
Is this still ongoing? I saw #1063 looks like abandoned?

**Comment by jfversluis** [comment]
This is still being worked on, thanks!

**Comment by albilaga** [comment]
Do you have any draft PR @jfversluis ? Want to check how is it going because in my implementation (based on #1063), some element can’t be clicked

**Comment by christianrr** [comment]
@jfversluis Some sort of roadmap would be great indeed, because I think native touch feedback is crucial for perceived responsiveness and state-of-the-art Apps.

We used TouchEffect.NativeAnimation heavily in our Xamarin.Forms applications and we need to know when this feature will be released for Maui in order to be able to estimate when we could release apps with Maui in the back.

**Comment by kkppstudios** [comment]
> @jfversluis Some sort of roadmap would be great indeed, because I think native touch feedback is crucial for perceived responsiveness and state-of-the-art Apps.
> 
> We used TouchEffect.NativeAnimation heavily in our Xamarin.Forms applications and we need to know when this feature will be released for Maui in order to be able to estimate when we could release apps with Maui in the back.

+1

**Comment by jfversluis** [comment]
I understand, it would be great to have that. However it's important to remember that this project is ran by people in their spare time. Sometimes we have more time, sometimes we have less time. Because of that, it's pretty hard to give any kind of estimate.

If it is vital or even critical to your project feel free to lend a hand, that way you will have full control over when you can use it in your project and seeing the engagement over this feature, you'll be making a lot of other people happy as well!

**Comment by christianrr** [comment]
@jfversluis I'll try my best. I've added an implementation for iOS (ported from XCT). I also fixed some bugs in the Android behavior.

For me, native feedback now works on both platforms.

#1326

**Comment by kphillpotts** [comment]
Nice work with the pull request @christianrr  - don't forget to agree to that Contributor License Agreement to progress

**Comment by kkohler2** [comment]
When will this be implemented?  I currently cannot update a Maui app with same functionality as Xamarin.Forms due to lack of this functionality.

**Comment by kkppstudios** [comment]
> When will this be implemented? I currently cannot update a Maui app with same functionality as Xamarin.Forms due to lack of this functionality.

Same here

**Comment by r-work** [comment]
There was [this](https://github.com/CommunityToolkit/Maui/pull/1326) pull request that was merged.
@kphillpotts Is there an ETA for when this may be available?

**Comment by kphillpotts** [comment]
The pull request was merged into the `touch-effect` branch which bought in iOS support. In that branch we now have Android, iOS and Windows, we still need to add in MacOS and Tizen and do provide the docs before release. I'm hoping in the next release.

**Comment by mirceamuresanse** [comment]
That's good news. I have a lot of respect for your real hard work. We all root for all of you who work so hard for this toolkit. 
According to Scrum principles, if Android, iOS and Windows are already tested and working, it would be nice to release it ASAP for these platforms, instead of waiting for all platforms to be implemented.

**Comment by VladislavAntonyuk** [comment]
@criss02-cs MauiCompat target framework is net6.0. 
most likely touch effect will be a part of the next release (with .net 8)

**Comment by mirceamuresanse** [comment]
Checking the code for the Touch Behavior in the touch-effect branch we noticed that the implementation is missing handling a move event and exposing it to the users. 
Are there any plans to implement the move event? 
This is critical for us, as we already use the Xamarin CommunityToolkit, where the TouchEffect is already handling the move event.

**Comment by kphillpotts** [comment]
At this point this proposal does not specify the move events.  It's a terrific addition no doubt to have this to maintain parity with Xamarin CommunityToolkit. I think it is best to get this version out as soon as we can and then we can put in the move events shortly after. 

If it is critical for you in the meantime you could port the XCT move across in a PR so we can get it in as soon as possible.

**Comment by criss02-cs** [comment]
approximately when will we have the opportunity to have this functionality?

**Comment by fredericoregateiro** [comment]
I do believe that some functionality is better than none.
To see some functionality from time to time is way better than years with nothing.

**Comment by Axemasta** [comment]
The `TouchEffect` in XCT in my opinion was a core feature that realistically should have been part of the main framework. It was the best part of XCT and all my Maui apps have been REALLY suffering in its absence.

The MauiCompat library doesn't seem to work in any of my net7 / 8 maui apps. I can see there is a planned release for this effect in net8 & can see the `touch-effect` branch looks quite good. I've hit the point where I simply couldn't wait for an MCT release so I have the XCT code ported to Maui [here](https://github.com/Axemasta/Maui.TouchEffect).

I'm not putting this on NuGet, for anyone wanting to use it my suggestion is build it locally and either host it in a private nuget feed (such as azure devops) or reference locally.

Really looking forward to this being released properly!

**Comment by gsgou** [comment]
@kphillpotts is there any rough estimation or planning when to expect this feature coming in?

**Comment by unamed000** [comment]
> I was watching .net conf and see this video on maui https://youtu.be/4saU9BNY6l4?t=677&si=8MkaeOaxJ8Edq7nk
> 
> They talk about point gestures and that they can be used to detect things like long press, also the docs https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/pointer?view=net-maui-8.0&tabs=windows
> 
> I didn't have the time to check this out, have anyone worked with these?

I don't think thing the Pointer gesture is enough, it doesn't have event that will be fired when user just hold on and didn't move their finger. So how can we detect something like that?

**Comment by fredericoregateiro** [comment]
@unamed000 
I didn't have time to try but in WPF i've made something similar with the down and up touch events, and we could use the [PointerPressed](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.pointergesturerecognizer.pointerpressed#microsoft-maui-controls-pointergesturerecognizer-pointerpressed) and [PointerReleased](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.pointergesturerecognizer.pointerreleased#microsoft-maui-controls-pointergesturerecognizer-pointerreleased) in MAUI.
So the idea is to start a timer on the PointerPressed event and check the passed time on the PointerReleased event.

It could work on all platforms with just one implementation.

**Comment by maexsp** [comment]
@fredericoregateiro Actually I use the package for TouchEffects unless its implemented in the CommunityToolkit. API is compatible to the old XF implementation and works fine for Android and iOS. Quite fresh nuget package but quality is ok. https://www.nuget.org/packages/TouchEffect.Maui

**Comment by Axemasta** [comment]
>  Quite fresh nuget package but quality is ok. https://www.nuget.org/packages/TouchEffect.Maui

This looks [familiar](https://github.com/felipebaltazar/TouchEffect.Maui/blob/2165a17938ffc4a360343254961daa3e8fef5046/samples/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample.csproj#L26C10-L26C10)... too [familiar](https://github.com/Axemasta/Maui.TouchEffect/blob/0f9b7a5f2c31307be23c3d2f41d3828b24b41903/samples/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample.csproj#L23C22-L23C22). Not that I put my maui port up for fame and glory, but when this library is clearly just my repo slightly reworded, with no credit or the crucial context that this library is a stopgap for the MCT feature, feels a little wild west...

Coincidentally I just updated my port to NET 8 & Fixed an iOS bug to do with input transparency (something I did after this repo was copied). We now have loads of wish.com maui touch effects floating around, we are in dire need of the official thing. Is there any word on the state of the current feature? I am super happy to jump in and help get this feature over the line so we can end the compat library wars!

**Comment by jfversluis** [comment]
@Axemasta 😔 yeah this is not great. Might be good to maybe get this out of the way, awesome you're willing to help out. Did you join our [Discord server](https://discord.gg/FYr7FWYhQk) by any chance? That might be good to get some coordination going?

**Comment by felipebaltazar** [comment]
> > Quite fresh nuget package but quality is ok. https://www.nuget.org/packages/TouchEffect.Maui
> 
> This looks [familiar](https://github.com/felipebaltazar/TouchEffect.Maui/blob/2165a17938ffc4a360343254961daa3e8fef5046/samples/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample.csproj#L26C10-L26C10)... too [familiar](https://github.com/Axemasta/Maui.TouchEffect/blob/0f9b7a5f2c31307be23c3d2f41d3828b24b41903/samples/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample/Maui.TouchEffect.Sample.csproj#L23C22-L23C22). Not that I put my maui port up for fame and glory, but when this library is clearly just my repo slightly reworded, with no credit or the crucial context that this library is a stopgap for the MCT feature, feels a little wild west...
> 
> Coincidentally I just updated my port to NET 8 & Fixed an iOS bug to do with input transparency (something I did after this repo was copied). We now have loads of wish.com maui touch effects floating around, we are in dire need of the official thing. Is there any word on the state of the current feature? I am super happy to jump in and help get this feature over the line so we can end the compat library wars!

@Axemasta First of all I want to apologize, you are right my package is entirely your code and I don't have any mention to you... 
I don't think that any explanation will be enough here, but I started forking from the original TouchEffect and i didn't notice that I didn't fork from your repository (and I don't put any mention in the readme, shame)

I don't used your Nuget because it's deprecated and you have mentioned that you have no intention to maintain other platforms, so I started my version to solve this for the app migrations that I'm doing... I want to talk to you in order to resolve this, maybe removing my package and contributing to your original code.. or anything that you have in mind...
Please text me: felipe.dasilvabaltazar@gmail.com

Sorry again

**Comment by IeuanWalker** [comment]
> > I was watching .net conf and see this video on maui https://youtu.be/4saU9BNY6l4?t=677&si=8MkaeOaxJ8Edq7nk
> > 
> > They talk about point gestures and that they can be used to detect things like long press, also the docs https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/pointer?view=net-maui-8.0&tabs=windows
> > 
> > I didn't have the time to check this out, have anyone worked with these?
> 
> I don't think thing the Pointer gesture is enough, it doesn't have event that will be fired when user just hold on and didn't move their finger. So how can we detect something like that? 

If u just want pressed, released and clicked events, you could try my control out - https://github.com/IeuanWalker/Maui.StateButton

**Comment by gsgou** [comment]
@felipebaltazar any plans for a PR to @Axemasta project to support Windows?
https://github.com/Axemasta/Maui.TouchEffect/pulls
Do you have any project with any branch open?

**Comment by alexskrms** [comment]
Any news about this feature coming to .NET MAUI Community Toolkit? I need it for my application.

**Comment by jfversluis** [comment]
Still being worked on, stay tuned!

**Comment by GeorgeCarr44** [comment]
> @unamed000 
> I didn't have time to try but in WPF i've made something similar with the down and up touch events, and we could use the [PointerPressed](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.pointergesturerecognizer.pointerpressed#microsoft-maui-controls-pointergesturerecognizer-pointerpressed) and [PointerReleased](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.pointergesturerecognizer.pointerreleased#microsoft-maui-controls-pointergesturerecognizer-pointerreleased) in MAUI.
> So the idea is to start a timer on the PointerPressed event and check the passed time on the PointerReleased event.
> 
> It could work on all platforms with just one implementation.

Your going to have issues with this implementation if the clickable element in within a scroll view and the user scrolls on the clickable element

**Comment by apoorvadixit-ttc** [comment]
> Still being worked on, stay tuned!

@jfversluis is there a short-term alternative for .NET 8/MAUI?

**Comment by jfversluis** [comment]
Or of course implement it yourself. Any plugin or project is someone else writing code you could've written yourself.

Otherwise, done when it's done 😄 and we hear you, we see a lot of people want this. So we're working on it.

**Comment by Axemasta** [comment]
@apoorvadixit-ttc 
> is there a short-term alternative for .NET 8/MAUI?

If you scroll up you'll see my links, I ported the XCT implementation to Maui and have it available as a "build, fork or drop into your projects yourself & use at your own risk" type deal. I'm currently using that library in my own apps and its been rock solid so far.

Currently working on the MCT PR to get this feature into the toolkit for 2 reasons. 1 its not my project, I just did a port and want the credit to go to the toolkit guys & Andrei who originally wrote it + donated to xct. 2 I dont want sole maintenance responsibility, it would be better when a team of people can manage it that way we know its super solid

**Comment by apoorvadixit-ttc** [comment]
> @apoorvadixit-ttc
> 
> > is there a short-term alternative for .NET 8/MAUI?
> 
> If you scroll up you'll see my links, I ported the XCT implementation to Maui and have it available as a "build, fork or drop into your projects yourself & use at your own risk" type deal. I'm currently using that library in my own apps and its been rock solid so far.
> 
> Currently working on the MCT PR to get this feature into the toolkit for 2 reasons. 1 its not my project, I just did a port and want the credit to go to the toolkit guys & Andrei who originally wrote it + donated to xct. 2 I dont want sole maintenance responsibility, it would be better when a team of people can manage it that way we know its super solid

Noted sir. I have tested it in a sample .NET 8/MAUI project and it seems to do what I was doing in the Xamarin CT, so thank you.
Hopefully we will see a new version of MAUI toolkit soon.

**Comment by jfversluis** [comment]
All done, released as part of [v8.0.0](https://github.com/CommunityToolkit/Maui/releases/tag/8.0.0)
